--- run.py.old	2025-01-10
+++ run.py	2025-01-10
@@ -100,7 +100,14 @@
     if hook_infos["action"] != "opened":
         api_logger.debug(
             "Received a pull_request webhook but doesn't qualify for starting a job."
         )
         return response.empty(status=204)

-    if hook_infos["pull_request"]["user"]["login"] not in ["github-actions[bot]", "yunohost-bot"] or not hook_infos["pull_request"]["head"]["ref"].startswith("ci-auto-update-"):
+    valid_user = (
+        app.config.EXECUTOR == ""
+        or hook_infos["pull_request"]["user"]["login"] == app.config.EXECUTOR
+    )
+    valid_ref = hook_infos["pull_request"]["head"]["ref"].startswith("ci-auto-update-")
+
+    if not valid_user or not valid_ref:
         api_logger.debug(
             "Received a pull_request webhook but from an unauthorized user or invalid ref."
         )
         return response.empty(status=204)
